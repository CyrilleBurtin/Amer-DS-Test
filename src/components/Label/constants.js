export const LABEL_VARIANT = {
  GRAY: 'gray'
}
