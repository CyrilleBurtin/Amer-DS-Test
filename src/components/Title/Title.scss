.slm-title {
  @apply text-salomon-black
    font-semibold
    font-inter;

  &.--red{
    @apply text-salomon-red;
  }

  &.--gray{
    @apply text-salomon-gray-400;
  }
}
