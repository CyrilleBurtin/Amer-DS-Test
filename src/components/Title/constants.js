export const TITLE_VARIANT = {
  RED: 'red',
  GRAY: 'gray'
};
