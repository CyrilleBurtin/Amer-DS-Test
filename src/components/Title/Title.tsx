import { ElementType, ReactNode } from 'react'
import { TITLE_VARIANT } from './constants'
import './Title.scss'
import { classNames } from '../../utils'

interface Title {
  is: ElementType
  children: string | ReactNode
  variant?: string
}

const Title = ({ is, variant, children, ...props }: Title) => {
  const TitleTag = is

  const classes = {
    title: classNames('slm-title', `--${variant}`)
  }

  return (
    <TitleTag className={classes.title} {...props}>
      {children}
    </TitleTag>
  )
}

export default Title
