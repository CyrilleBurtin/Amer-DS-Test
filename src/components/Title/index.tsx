import { ReactHTML, ReactNode } from 'react'
import './Title.scss'
import { classNames } from '../../utils'

interface Title {
  is: keyof ReactHTML
  children: string | ReactNode
  variant?: string
  className?: string
}

const Title = ({ is, variant = 'gray', children, className }: Title) => {
  const TitleTag = is

  const classes = {
    title: classNames('slm-title', className, `--${variant}`)
  }

  return <TitleTag className={classes.title}>{children}</TitleTag>
}

export default Title
