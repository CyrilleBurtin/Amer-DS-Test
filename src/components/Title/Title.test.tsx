import { render } from '@testing-library/react';
import Title from './index.js';
import { TITLE_VARIANT } from './constants';

describe('<Title>', () => {
  it('should render the default component with custom text', () => {
    const { container, getByText } = render(
      <Title is="h2" variant={TITLE_VARIANT.RED}>
        Custom Title
      </Title>
    );

    expect(container.firstChild).toHaveClass('slm-title');
    expect(getByText(/Custom title/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the default component with custom text and custom class', () => {
    const { container, getByText } = render(
      <Title is="h1" className="p-4">
        Custom title
      </Title>
    );

    expect(container.firstChild).toHaveClass('slm-title p-4');
    expect(getByText(/Custom title/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
