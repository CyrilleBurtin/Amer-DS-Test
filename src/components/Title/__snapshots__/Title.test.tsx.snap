// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Title> > should render the default component with custom text 1`] = `
<div>
  <h2
    class="slm-title --red"
  >
    Custom Title
  </h2>
</div>
`;

exports[`<Title> > should render the default component with custom text and custom class 1`] = `
<div>
  <h1
    class="slm-title p-4 --gray"
  >
    Custom title
  </h1>
</div>
`;
